packages/web-api/src/WebClient.ts

```diff
@@ -388,6 +393,143 @@ export class WebClient extends Methods {
+
+    // 3
+    await this.postCompletedFileUploads(fileUploads);
+
+    // 4
+    const res = await this.getFileInfo(fileUploads);
+
+    return { ok: true, data: res };
+  }
+
+  /**
+   * For each file submitted with this method, submits filenames
+   * and file metadata to files.getUploadURLExternal to request a URL to
+   * which to send the file data to and an id for the file
+   * @param fileUploads
+   * @returns
+   */
+  private async fetchAllUploadURLExternal(fileUploads: FileUploadV2Entry[]):
+  Promise<Array<FilesGetUploadURLExternalResponse>> {
+    return Promise.all(fileUploads.map((upload: FileUploadV2Entry) => {
+      /* eslint-disable @typescript-eslint/consistent-type-assertions */
+      const options = {
+        filename: upload.filename,
+        length: upload.length,
+        alt_text: upload.alt_text,
+        snippet_type: upload.snippet_type,
+      } as FilesGetUploadURLExternalArguments;
+
+      return this.files.getUploadURLExternal(options);
+    }));
+  }
+
+  /**
+   * For each successful upload, completeUploadExternal endpoint,
+   * with a list of completed uploads - opt. channel id, initial comment, thread_ts
+   * @param fileUploads
+   * @returns
+   */
+  private async postCompletedFileUploads(fileUploads: FileUploadV2Entry[]):
+  Promise<Array<WebAPICallResult>> {  
+    const jobs = getAllFileUploadsToComplete(fileUploads);
+    const jobsValues = Object.values(jobs);
+
+    return Promise.all(
+      jobsValues.map((job: FilesCompleteUploadExternalArguments) => this.files.completeUploadExternal(job)),
+    );
+  }
+
+  private async getFileInfo(fileUploads: FileUploadV2Entry[]):
+  Promise<Array<WebAPICallResult>> {
+    return Promise.all(fileUploads.map((upload: FileUploadV2Entry) => this.files.info({ file: upload.file_id! })));
+  }
+
+  /**
+   * for each returned file upload URL, upload corresponding file
+   * @param fileUploads
+   * @returns
+   */
+  private async postFileUploadsToExternalURL(fileUploads: FileUploadV2Entry[], options: FilesUploadV2Arguments)
+    : Promise<Array<FilesGetUploadURLExternalResponse>> {
+    return Promise.all(fileUploads.map(async (upload: FileUploadV2Entry) => {
+      const { upload_url, file_id, filename, data } = upload;
+      // either file or content will be defined
+      const body = data;
+
+      // try to post to external url
+      if (upload_url) {
+        const headers: Record<string, string> = {};
+        if (options.token) headers.Authorization = `Bearer ${options.token}`;
+
+        const uploadRes = await this.makeRequest(upload_url, {
+          body,
+        }, headers);
+        if (uploadRes.status !== 200) {
+          return Promise.reject(Error(`Failed to upload file id:${file_id}, ${filename}`));
+        } 
+        const returnData = { ok: true, body: uploadRes.data } as WebAPICallResult;
+        return Promise.resolve(returnData);
+      } 
+      return Promise.reject(Error('No upload url found for file'));
```

Comment:
```suggestion
      return Promise.reject(Error(`No upload URL found for a file (id: ${file_id}, name: ${filename})`));
```

Reply:
There won't be a file_id in this case, so I have added the file name portion only!